text/0000-pibd-messages.md

```diff
@@ -0,0 +1,128 @@
+
+- Title: pibd-messages
+- Authors: [Jasper van der Maarel](mailto:j@sper.dev), [Antioch Peverell](mailto:apeverell@protonmail.com)
+- Start date: September 14, 2020
+- RFC PR: Edit if merged: [mimblewimble/grin-rfcs#0000](https://github.com/mimblewimble/grin-rfcs/pull/0000) 
+- Tracking issue: [Edit if merged with link to tracking github issue]
+
+---
+
+# Summary
+[summary]: #summary
+
+TODO 
+
+# Motivation
+[motivation]: #motivation
+
+TODO
+
+# Community-level explanation
+[community-level-explanation]: #community-level-explanation
+
+TODO
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+TODO
+
+## MMR chunks
+[mmr-chunks]: #mmr-chunks
+We define a (P)MMR **chunk** as a set of `N = 2^b` consecutive leaves, along with the necessary data to reconstruct the subtree root and data to verify membership of the subtree in the original MMR.
+Each chunk is consecutive with the previous and is identified by the tuple `(H, b, i)` where `H` is the block hash and `i` is a zero-based chunk index: chunk `(H, b, i)` contains the non-pruned leaves with leave position in the interval `[i*N, (i+1)*N)`.
+The final chunk is allowed to have less than `N` elements because the full MMR does not necessarily contain a multiple of `N` leaves.
+
+A chunk contains, besides the identifier tuple and the unpruned leaf data, a list of pruned hashes of the leaves and a chunk MMR proof.
```

Comment:
"a list of pruned hashes of the leaves" is misleading. It's not the hashes of the leaves, but instead the least-height unpruned parent hashes of the pruned leaves. I don't know how to word that unambiguously though 😆

Reply:
You are right, I will try to word it more accurately.